js/krakenfutures.js

```diff
@@ -0,0 +1,2032 @@
+        //          },
+        //          ...
+        //        ]
+        //    }
+        //
+        const rates = this.safeValue (response, 'rates');
+        const result = [];
+        for (let i = 0; i < rates.length; i++) {
+            const item = rates[i];
+            const datetime = this.safeString (item, 'timestamp');
+            result.push ({
+                'info': item,
+                'symbol': symbol,
+                'fundingRate': this.safeNumber (item, 'fundingRate'),
+                'timestamp': this.parse8601 (datetime),
+                'datetime': datetime,
+            });
+        }
+        const sorted = this.sortBy (result, 'timestamp');
+        return this.filterBySymbolSinceLimit (sorted, symbol, since, limit);
+    }
+
+    async fetchPositions (symbols = undefined, params = {}) {
+        /**
+         * @method
+         * @name krakenfutures#fetchPositions
+         * @description Fetches current contract trading positions
+         * @param {[string]} symbols List of unified symbols
+         * @param {object} params Not used by krakenfutures
+         * @returns Parsed exchange response for positions
+         */
+        await this.loadMarkets ();
+        const request = {};
+        const response = await this.privateGetOpenpositions (request);
+        //
+        //    {
+        //        result: 'success',
+        //        openPositions: [
+        //            {
+        //                side: 'long',
+        //                symbol: 'pi_xrpusd',
+        //                price: '0.7533',
+        //                fillTime: '2022-03-03T22:51:16.566Z',
+        //                size: '230',
+        //                unrealizedFunding: '-0.001878596918214635'
+        //            }
+        //        ],
+        //        serverTime: '2022-03-03T22:51:16.566Z'
+        //    }
+        //
+        const result = this.parsePositions (response);
+        return this.filterByArray (result, 'symbol', symbols, false);
+    }
+
+    parsePositions (response, symbols = undefined, params = {}) {
+        const result = [];
+        const positions = this.safeValue (response, 'openPositions');
+        for (let i = 0; i < positions.length; i++) {
+            const position = this.parsePosition (positions[i]);
+            result.push (position);
+        }
+        return result;
+    }
+
+    parsePosition (position, market = undefined) {
+        //
+        //    {
+        //        side: 'long',
+        //        symbol: 'pi_xrpusd',
+        //        price: '0.7533',
+        //        fillTime: '2022-03-03T22:51:16.566Z',
+        //        size: '230',
+        //        unrealizedFunding: '-0.001878596918214635'
+        //    }
+        //
+        const datetime = this.safeString (position, 'fillTime');
+        return {
+            'info': position,
+            'symbol': this.safeCurrencyCode (this.safeString (position, 'symbol')),
```

Comment:
I was playing around with this. I noticed the symbol that ends up in the dict is the symbol from Kraken (market id), but not the unified one. I think it should be taken from the market data, something like:
`
market = this.safeMarket (this.safeString (position, 'symbol'), market);
# Then use market['symbol'];`